################################################################################
SHORT DESCRIPTION: 
################################################################################
Port git://xenbits.xen.org/xen.git:0ff01eebcd72f3e750e8659a5e14b3bd20744b7b

################################################################################
LONG DESCRIPTION: 
################################################################################
commit 0ff01eebcd72f3e750e8659a5e14b3bd20744b7b
Author: Frediano Ziglio <frediano.ziglio@citrix.com>
Date:   Mon May 7 13:39:27 2012 +0100

    vgabios: Make Windows 8 support greater resolutions
    
    Apparently Windows 8 refuse to use any mode if has more than one page.
    
    Signed-off-by: Frediano Ziglio <frediano.ziglio@citrix.com>
    Committed-by: Keir Fraser <keir@xen.org>


################################################################################
CHANGELOG 
################################################################################
Ported to VGABIOS: Eric Chanudet, chanudete@ainfosec.com, 01/03/2015

################################################################################
REMOVAL 
################################################################################
This patch can be removed if a new version of Bochs VGABIOS is released with
this fix.

################################################################################
UPSTREAM PLAN 
################################################################################
This is already upstream in Xen repository.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None.

################################################################################
PATCHES
################################################################################
--- a/vbe.c
+++ b/vbe.c
@@ -957,9 +957,9 @@ Bit16u *AX;Bit16u CX; Bit16u ES;Bit16u D
                     (size_64k > totalMemory))
                   info.ModeAttributes &= ~VBE_MODE_ATTRIBUTE_SUPPORTED;
 
-                if (using_lfb) {
-                  info.NumberOfBanks = 1;
-                }
+                /* Windows 8 require this to be 1! */
+                info.NumberOfBanks = 1;
+
 #ifdef PCI_VID
                 lfb_addr = pci_get_lfb_addr(PCI_VID);
 #else
